lib/active_merchant/billing/gateways/netbanx.rb

```diff
@@ -5,16 +5,25 @@ class NetbanxGateway < Gateway
       self.test_url = 'https://api.test.netbanx.com/'
       self.live_url = 'https://api.netbanx.com/'
 
-      self.supported_countries = ['CA', 'US', 'GB']
+      self.supported_countries = %w(AF AX AL DZ AS AD AO AI AQ AG AR AM AW AU AT AZ BS BH BD BB BY BE BZ BJ BM BT BO BQ BA BW BV BR IO BN BG BF BI KH CM CA CV KY CF TD CL CN CX CC CO KM CG CD CK CR CI HR CU CW CY CZ DK DJ DM DO EC EG SV GQ ER EE ET FK FO FJ FI FR GF PF TF GA GM GE DE GH GI GR GL GD GP GU GT GG GN GW GY HT HM HN HK HU IS IN ID IR IQ IE IM IL IT JM JP JE JO KZ KE KI KP KR KW KG LA LV LB LS LR LY LI LT LU MO MK MG MW MY MV ML MT MH MQ MR MU YT MX FM MD MC MN ME MS MA MZ MM NA NR NP NC NZ NI NE NG NU NF MP NO OM PK PW PS PA PG PY PE PH PN PL PT PR QA RE RO RU RW BL SH KN LC MF VC WS SM ST SA SN RS SC SL SG SX SK SI SB SO ZA GS SS ES LK PM SD SR SJ SZ SE CH SY TW TJ TZ TH NL TL TG TK TO TT TN TR TM TC TV UG UA AE GB US UM UY UZ VU VA VE VN VG VI WF EH YE ZM ZW)
       self.default_currency = 'CAD'
-      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+      self.supported_cardtypes = [
+        :american_express,
+        :diners_club,
+        :discover,
+        :jcb,
+        :master,
+        :maestro,
+        :swiff,
+        :visa,
+        :visa_debit,
+        :visa_electron
+      ]
       self.money_format = :cents
 
       self.homepage_url = 'https://processing.paysafe.com/'
       self.display_name = 'Netbanx by PaySafe'
 
-      STANDARD_ERROR_CODE_MAPPING = {}
```

Comment:
Hey @iirving - I've taken a quick look at the list of error codes and those look like they fit the bill. A possible way to implement this could be to map highly specific error codes (e.g. AVS/CVC checks) in this constant and use the example [`error_code_from(response)`](https://github.com/activemerchant/active_merchant/blob/b2f62a950faab76d1e0417fd090602ae5ac3363b/generators/gateway/templates/gateway.rb#L116) method to look up the code here and return it.

If it is not found, apply some heuristics to figure out if `:card_declined` (e.g. `if response.code.to_s.start_with?('30')`) and fall back to the generic `:processing_error`. Hope this helps!